File: Java/Eta/ValueAdd/src/main/java/com/refinitiv/eta/valueadd/reactor/RDMDictionaryMsgEvent.java
/*|-----------------------------------------------------------------------------
 *|            This source code is provided under the Apache 2.0 license
 *|  and is provided AS IS with no warranty or guarantee of fit for purpose.
 *|                See the project's LICENSE.md for details.
 *|           Copyright (C) 2020,2022,2024 LSEG. All rights reserved.
 *|-----------------------------------------------------------------------------
 */

package com.refinitiv.eta.valueadd.reactor;

import com.refinitiv.eta.valueadd.domainrep.rdm.dictionary.DictionaryMsg;

/**
 * Event provided to RDMDictionaryMsgCallback methods.
 * 
 * @see ReactorMsgEvent
 */
public class RDMDictionaryMsgEvent extends ReactorMsgEvent
{
    DictionaryMsg _dictionaryMsg;

    RDMDictionaryMsgEvent()
    {
        super();
    }

    void rdmDictionaryMsg(DictionaryMsg rdmDictionaryMsg)
    {
        _dictionaryMsg = rdmDictionaryMsg;
    }

    /**
     * The DictionaryMsg associated with this message event.
     * 
     * @return DictionaryMsg
     */
    public DictionaryMsg rdmDictionaryMsg()
    {
        return _dictionaryMsg;
    }
}

File: CSharp/Eta/Src/ValueAdd/Reactor/RDMDictionaryMsgEvent.cs
/*|------------------------------------------get; set;-----------------------------------
 *|            This source code is provided under the Apache 2.0 license      --
 *|  and is provided AS IS with no warranty or guarantee of fit for purpose.  --
 *|                See the project's LICENSE.md for details.                  --
 *|           Copyright (C) 2022-2023 Refinitiv. All rights reserved.         --
 *|-----------------------------------------------------------------------------
 */

using LSEG.Eta.ValueAdd.Rdm;

namespace LSEG.Eta.ValueAdd.Reactor
{
    /// <summary>
    /// Event provided to <see cref="IDictionaryMsgCallback"/> methods.
    /// </summary>
    public class RDMDictionaryMsgEvent : ReactorMsgEvent
    {
        /// <summary>
        /// Gets the <see cref="DictionaryMsg"/> associated with this message event.
        /// </summary>
        public DictionaryMsg? DictionaryMsg { get; internal set; }

        /// <summary>
        /// 
        /// </summary>
        public override void Clear()
        {
            base.Clear();
            DictionaryMsg = null;
        }

        /// <summary>
        /// 
        /// </summary>
        public override void ReturnToPool()
        {
            DictionaryMsg = null;
            base.ReturnToPool();
        }
    }
}
